server/src/main/java/io/spine/server/bus/Bus.java

```diff
@@ -71,18 +88,106 @@ public void unregister(D dispatcher) {
+    }
+
+    /**
+     * Posts the given messages to the bus.
+     *
+     * <p>Use the {@code Bus} class abstract methods to modify the behavior of posting.
+     *
+     * <p>The {@link StreamObserver} argument is the posting outcome observer;
+     * the {@link StreamObserver#onNext StreamObserver.onNext} will be called for each message
+     * passed to the bus.
+     *
+     * <p>A valid message will result in
+     * an {@link io.spine.base.Status.StatusCase#OK OK} status {@link IsSent} instance.
+     *
+     * <p>An invalid message will result in an {@link io.spine.base.Error Error} status
+     * {@link IsSent} instance.
+     *
+     * <p>Depending on the underlying {@link MessageDispatcher}, a message which causes a business
+     * {@link io.spine.base.Failure} may result ether a {@link io.spine.base.Failure} status or
+     * an {@link io.spine.base.Status.StatusCase#OK OK} status {@link IsSent} instance. Usually,
+     * the {@link io.spine.base.Failure} status may only pop up if the {@link MessageDispatcher}
+     * processes the message sequentially and throws
+     * the {@linkplain io.spine.base.FailureThrowable FailureThrowables} (wrapped in a
+     * {@link RuntimeException}) instead of handling them. Otherwise, the {@code OK} status should
+     * be expected.
+     *
+     * <p>Note that the {@code observer} is always positive, i.e.
+     * {@link StreamObserver#onError StreamObserver.onError()} will never be called.
+     *
+     * @param messages the message to post
+     * @param observer the observer to receive outcome of the operation
      */
-    public abstract void post(T message, StreamObserver<Response> responseObserver);
+    public final void post(Iterable<T> messages, StreamObserver<IsSent> observer) {
+        checkNotNull(messages);
+        checkNotNull(observer);
+
+        final Iterable<T> filteredMessages = filter(messages, observer);
+        if (!isEmpty(filteredMessages)) {
+            store(messages);
+            final Iterable<E> envelopes = transform(filteredMessages, toEnvelope());
+            doPost(envelopes, observer);
+        }
+        observer.onCompleted();
+    }
+
+    /**
+     * Acknowledges the sent envelope.
+     *
+     * @param envelope the envelope to acknowledge
+     * @return the envelope acknowledgement
+     */
+    public final IsSent acknowledge(E envelope) {
+
+        return setStatus(envelope, Responses.statusOk());
+    }
+
+    /**
+     * Sets the given status to the sent envelope.
+     *
+     * @param envelope the envelope to provide with a status
+     * @param status   the status of the envelope
+     * @return the envelope posting result
+     */
+    public final IsSent setStatus(E envelope, Status status) {
+        checkNotNull(envelope);
+        checkNotNull(status);
+        checkArgument(isNotDefault(status));
+
+        final Message id = getId(envelope);
+        final Any packedId = pack(id);
+        final IsSent result = IsSent.newBuilder()
```

Comment:
This is a duplicate. See your changes to `CommandService`. As long as `IsSent` is in the `bus` package, maybe we could create `Buses` utility class and have a factory utility to produce the instances?